### PR TITLE
Bleed page improvements + pack logic part 2

### LINKED ISSUE #1088: [BUG] Missing pack 1/2 "fault" light with no bleed air
**Mod Version**

Latest

**Describe the bug**

When the Pack 1 and Pack 2 pushbuttons are pushed in, they are in the "auto" position. This means that if there is no bleed air to the Packs, the "fault" light should illuminate on the Pack pushbuttons because they're not on.

**To Reproduce**

1. Connect airplane to ground power or start APU without turning on the APU bleed.
2. Ensure the PACK 1 and PACK 2 pushbuttons are pushed (in auto mode, not off).
3.

**Expected behavior**

Due to the pushbuttons for the Packs being set to the "auto" mode (no off light), the "fault" light should illuminate if there is no bleed air to the system because the packs aren't on in this state.

**Actual behavior**

Button has no "fault" light even without bleed air.

**References**

Here is the FFA320 in XP11 with the APU on, APU bleed off, PACK 1 and 2 pushbuttons in auto mode. Notice the fault lights are illuminated due to lack of bleed air:
A320 - 2020-10-04 11 08 37

Here is the FBW a32nx in a similar state. The first screenshot shows the lack of bleed pressure on the ECAM and the second screenshot shows the missing PACK "fault" lights:
Screenshot (75)
Screenshot (76)

**Additional context**

Was this working before/when did the issue start occurring?

Is this a problem in the vanilla unmodded game?

Discord username (if different from GitHub):

### LINKED ISSUE #1263: [BUG] ECAm bleed page seems a bit messed up
**Mod Version**

current master

**Describe the bug**
After implementing PR #1238 the bleed page seems to be a bit messed up.

**To Reproduce**

1. Enter cold&dark
2. follow the normal procedures to start the engines
3. turn the air cond temps to 12 o'clock
4. when engines running switch to ECAM bleed page

**Expected behavior**
Lower temps on the outflow. A single line/pointer below. 

**Actual behavior**
see below

**References**
bleedpage

**Additional context**

Was this working before/when did the issue start occurring?
see above

### PR DESCRIPTION
**Pull request by nikotina91**
Fixes #1088
Fixes #1263

## Summary of Changes
Did more work on the bleed page, still a work in progress, changes:

- Engine bleed connections (IP/HP) show in amber if there's no pressure available
- HP valves close when engine N2 rpm is slightly above idle, only IP valves remain open 
- Lines on the page should now disappear correctly based on valve position
- "GND" and respective symbol now disappear when in flight
- "APU", APU valve and connection now disappear when the APU is off (? not sure about this one)
- Shown values turn amber when out of range
   - Pack outflow turns amber after 90°
   - Pack inflow turn amber after 230°
   - pressures turn amber when below 4 psi or above 57 psi
   - temperatures turn amber when below 150° or above 270°
   - engine numbers turn amber when N2 below idle and FADEC is energized
- All the temperatures should now be more inline with reality (needs more testing)
- APU bleed temperatures based on APU N2 rpm
- Fault lights now turn on when the packs don't have pressure but the valve is open (issue #1088)
- Packs close during engine start
- Pack are forced to high flow when:
   - TO/GA or MCT/FLX power are applied
   - APU provides bleed
   - Single pack operations
- RAM AIR valve closes when takeoff power is applied
- Pack outlines are now thinner
- "PSI" and "C°" are now in white instead of blue

## Screenshots (if necessary)
All valves open / throttle idle / APU on / on the ground:
screen1

Same as before but xbleed is closed:
screen2

APU off and xbleed closed:
screen3

APU off and xbleed open:
screen4

xbleed closed, APU powering up:
screen5

eng1 on idle and eng1 bleed open / APU on APU bleed closed:
screen6

RAM AIR CLOSED, packs have no pressure + fault lights:
screen7
screen8

eng 1 bleed on / eng 2 bleed off / apu bleed off / xbleed on / right pack fault:
screen9

engines powering down:
screen20

fadec powered and N2 below idle:
fadec_powered_engine_below_idle_amber

fadec unpowered and N2 below idle:
fadec_unpowered_eng_below_idle_white

Not sure about this fadec powered/unpowered conditon, will need feedback

## References
cond and packs manuel about the a320: 
http://www.smartcockpit.com/docs/A320-Air_Conditioning_and_Pressurization.pdf

screen10
screen11
screen12
screen13

http://www.smartcockpit.com/docs/A320-Pneumatic.pdf
screen14
screen15
screen16

discord:
screen17
screen18
screen19

## Additional context
It still needs some tuning and probably bug fixing

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build.py will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, slick on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by FlyingCarpet7** [comment]
Awesome work! Well done!
I will try to give you irl reference  next week on, 1,2- colours, pack ram air operation and som other things. Will take some pictures.

**Comment by nikotina91** [comment]
thanks! that will be very useful!

**Comment by nikotina91** [comment]
Will probably need to change a few things when PR #1238 is merged.

**Comment by nikotina91** [comment]
The pack indicators will only move a few degrees without having the continuous aircond knobs PR as the value of the knob previously were in a 0-2 range and with it they are in a 0-100 range. So please if you are testing this download that PR as well or keep in mind why the indicators will not move much, everything should be fine.

**Comment by Carter633** [comment]
Name: British Avgeek YT - Carter633#1436
Date of testing: Current date 12/10/2020
Version of the sim: 1.9.3.0
PR Tested: #1209

Testing technique: Full flight from Cold and Dark
Steps you've made to test the Pull Request:
1. Spawned at Zurich, Ground power, checked bleed page
2. Started APU - reviewed bleed page throughout
3. Started engines - reviewed bleed page throughout
4. Took off and changed temp settings

Overall Rating: Good
Conclusions of the testing: No issues found

**Comment by lukecologne** [comment]
Small problem: The Pack temperature is clipping into the "°C", I don't think that is specific to your PR, but it would be cool if you could fix that. I think it was introduced after the font update, and it looks like the font size may be too big
grafik

**Comment by nikotina91** [comment]
yeah its the new font, give me a bit and I'll fix it, will also take the chance to add the anti ice indication as suggested by FlyingCarpet7 (with reference pics)

**Comment by nikotina91** [comment]
- Added anti-ice indicators
ref2

indicators turn amber when there's no bleed supply: https://www.ipadecam.co.uk/#
ref4

- fixed pack temperatures clipping into °C

- changed some of the text back to blue (this pic is also from FlyingCarpet7)
ref1

- ram air valve stays close when the pressure difference is more than one psi
image

This is what it looks it like now:
image

**Comment by lukecologne** [comment]
I promise, this is my last one: can you reduce the .st9 stroke width a bit (to around 2)? IMO the arc lines and GND triangles are a little to thick

**Comment by nikotina91** [comment]
image

**Review by lukecologne — APPROVED** [review]
Tested and LGTM. I didn't test the systems logic in-depth, since those will hopefully be soon™ implemented in C++.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html [hunk 1]
```diff
@@ -6,255 +6,260 @@
...
-			<!--engine numbers-->
-			<text transform="matrix(1 0 0 1 66.1666 414.9644)" class="st1 st2 st6">1</text>
-			<text transform="matrix(1 0 0 1 546.1666 414.9644)" class="st1 st2 st6">2</text>	
+			<!--title-->
 			
-			<!--LO/HI next to packs-->
-			<text transform="matrix(1 0 0 1 411.1766 214.3244)" class="st1 st2 st6">L</text>
-			<text transform="matrix(1 0 0 1 420.3266 214.3244)" class="st1 st2 st6">O</text>
-			<text transform="matrix(1 0 0 1 543.1766 214.3244)" class="st1 st2 st6">HI</text>
+			<text x="63" y="33" t class="st1 s2 st3">BLEED</text>
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:11]:
```suggestion
			<text x="63" y="33" class="st1 s2 st3">BLEED</text>
```
Pretty sure this shouldn't be there, sorry I didn't find it earlier

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html [hunk 2]
```diff
@@ -6,255 +6,260 @@
 
 <!-- <rect y="0" width="600" height="600" style="fill:url(#ECAMBacklight);"/> -->
 		<g>
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:8]: Either a missing end </g> tag, or the <g> should be removed

**Review by lukecologne — CHANGES_REQUESTED** [review]
Sorry I didn't find these earlier

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html [hunk 3]
```diff
@@ -6,255 +6,260 @@
...
+					<text transform="matrix(1.22 0 0 1 65 268)" style="font-size: 18px;" class="st1 st2">ANTI</text>
+					<text transform="matrix(1.22 0 0 1 535 268)" style="font-size: 18px;" class="st1 st2">ANTI</text>
+					<text transform="matrix(1.22 0 0 1 58 285)"  style="font-size: 18px;"class="st1 st2">ICE</text>
+					<text transform="matrix(1.22 0 0 1 540 285)" style="font-size: 18px;" class="st1 st2">ICE</text>
+				</g>
+			</g>
 			<!--temp boxes-->
-			<rect x="98.22" y="317.54" class="st10" width="43" height="49" />
-			<rect x="458.72" y="318.04" class="st10" width="43" height="49" />
+			<rect class="st12" x="100" y="317" class="st10" width="42" height="49" />
```
Comment (lukecologne) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:125]: Duplicate class attribute

**Review by lukecologne — APPROVED** [review]
Alright, I fixed the issues I pointed out, and also improved the indentation. I hope this was OK for you.

**Pull request merged**